Cubical/Categories/Site/Instances/ZariskiCommRing.agda

```diff
@@ -105,3 +114,117 @@ pullbackStability zariskiCoverage {c = A} um {d = B} φ =
 
   ψComm : ∀ i → (ψ i .fst) ∘ (AU._/1 (um .f i)) ≡ (BU._/1 (φ $r um .f i)) ∘ φ .fst
   ψComm i = uniqInvElemHom φ (um .f i) .fst .snd
+
+
+-- this defines a subcanonical coverage
+-- the proof is analoguous to the case of lattice sheaves,
+-- see "isLimConeLocCone" in Cubical.Algebra.CommRing.Localisation.Limit
+-- first, a lemma
+module SubcanonicalLemmas (A R : CommRing ℓ) where
+  open CommRingStr ⦃...⦄
+  open CommIdeal R using (_∈_)
+
+  private instance
+   _ = A .snd
+   _ = R .snd
+
+  module _
+    {n : ℕ}
+    (f : FinVec ⟨ R ⟩ (suc n)) -- have to treat case of empty vec separately
```

Comment:
I tried to understand why the n = 0 case should be treated separately, and so far I am not convinced. ;-)

The `equalizerLemma` that you use is formulated for `suc n` instead of general `n`. But I just went through the whole file `Localization/Limit.agda` where it is proven and it turns out that in all of these 600 lines, `(suc n)` can simply replaced by `n` and it still works, except for the one application of `invElPropElimN`! And when I look at `invElPropElimN` (in `InvertingElements.agda`) it looks like one could just start the induction there at 0 instead of 1 and it should work (unless I am missing something).

Reply:
Awesome, should we merge that PR first? Then I can do another rebase and simplify the proof.

Reply:
It worked!